metrics/cli/docs/02_enabling_firebase_analytics.md

```diff
@@ -0,0 +1,72 @@
+
+Let's review the pros and cons of this option.
+
+Pros:
+
+- easy to implement;
+
+- provides simple and understandable steps for users (visit the printed link and click the `Enable Google Analytics` button).
+
+Cons:
+
+- does not allow us to deploy the application from the CIs since it requires user interaction.
+
+#### Using an API
+
+The second approach is to call an [endpoint](https://firebase.google.com/docs/projects/api/reference/rest/v1beta1/projects/addGoogleAnalytics) from the [Firebase Management API](https://firebase.google.com/docs/projects/api/reference/rest), which enables Firebase Analytics for the current Firebase project.
+
+The endpoint's request body described above requires an `analytics account id`, which we can get from the [`Google analytics summaries endpoint`](https://developers.google.com/analytics/devguides/config/mgmt/v3/mgmtReference/management/accountSummaries/list).
+
+The `Google analytics summaries enpoint` requires a token with an `analytics.edit` scope. 
+
+Here is a table, which describes how to get the token with a specific scope:
+
+| Solution | **Description** | **Problems** |
+| --- | --- | --- |
+| **[Configure OAuth 2.0](https://support.google.com/cloud/answer/6158849) for the current GCloud project** | Allows us to get the token with required scope using OAuth 2.0 | The user should manually configure an [OAuth consent screen](https://support.google.com/cloud/answer/10311615) |
+| **Enable [compute engine API](https://cloud.google.com/compute/docs/reference/rest/v1) in GCloud account** | Allow us to give a required scope for Gcloud project service accounts using the command described [here](https://cloud.google.com/compute/docs/access/create-enable-service-accounts-for-instances). Then we can [authenticate](https://cloud.google.com/sdk/gcloud/reference/auth/activate-service-account) using the service account with the required scope and [print the service account token](https://cloud.google.com/sdk/gcloud/reference/auth/print-access-token). | The compute engine API requires enabling the Gcloud billing account, which a manually step only |
+
+Pros:
+
+- provides an ability to ensure the Firebase Analytics is enabled, and we can proceed with the deployment process.
+
+Cons: 
+
+- complicated implementation;
+- requires complicated steps from the user side (configure OAuth consent screen, configure billing account);
+- does not allow automating the Firebase Analytics enabling process.
+
+### Decision
+
+As we've analyzed above, the [API method](#using-an-api) does not allow us to automate the Firebase Analytics enabling process for now, so
+we have to choose the described [manual method](#manual) to enable Firebase Analytics because the manual steps is easier and faster for the user.
```

Comment:
```suggestion
As we've analyzed above, the [API method](#using-an-api) does not allow us to automate the Firebase Analytics enabling process, so
we are going to use the [manual method](#manual) since it provides a more clean and fast way of enabling Firebase Analytics.
```
?